pipeline/barrier.go

```diff
@@ -0,0 +1,43 @@
+package pipeline
+
+import (
+	"fmt"
+	"time"
+)
+
+// A BarrierNode will periodically emit a barrier with the current time, according to
+// the system clock.  Since the BarrierNode emits based on system time, it allows
+// pipelines to be forced in the absence of data traffic.
+//
+// Example:
+//    stream
+//        |barrier().every(5s)
+//        |window()
+//            .period(10s)
+//            .every(5s)
+//        |top('value', 10)
+//        //Post the top 10 results over the last 10s updated every 5s.
+//        |httpPost('http://example.com/api/top10')
+//
+type BarrierNode struct {
+	chainnode
+
+	// How often a barrier is emitted into the pipeline.
+	// Must be greater than zero.
+	Every time.Duration
```

Comment:
We should add a comment there that the time is real time, not data time. I know you have it above but it would be good to reiterate here as well.